hoodie-utilities/src/main/java/com/uber/hoodie/utilities/HoodieDataImporter.java

```diff
@@ -0,0 +1,236 @@
+        SparkConf sparkConf = new SparkConf().setAppName("hoodie-data-importer-" + cfg.tableName);
+        sparkConf.setMaster(cfg.sparkMaster);
+
+        if (cfg.sparkMaster.startsWith("yarn")) {
+            sparkConf.set("spark.eventLog.overwrite", "true");
+            sparkConf.set("spark.eventLog.enabled", "true");
+        }
+
+        sparkConf.set("spark.driver.maxResultSize", "2g");
+        sparkConf.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer");
+        sparkConf.set("spark.executor.memory", cfg.sparkMemory);
+
+        // Configure hadoop conf
+        sparkConf.set("spark.hadoop.mapred.output.compress", "true");
+        sparkConf.set("spark.hadoop.mapred.output.compression.codec", "true");
+        sparkConf.set("spark.hadoop.mapred.output.compression.codec",
+            "org.apache.hadoop.io.compress.GzipCodec");
+        sparkConf.set("spark.hadoop.mapred.output.compression.type", "BLOCK");
+
+        sparkConf = HoodieWriteClient.registerClasses(sparkConf);
+        return new JavaSparkContext(sparkConf);
+    }
+
+    private String getSchema(FileSystem fs) throws Exception {
+        // Read schema file.
+        Path p = new Path(cfg.schemaFile);
+        if (!fs.exists(p)) {
+            throw new Exception(
+                String.format("Could not find - %s - schema file.", cfg.schemaFile));
+        }
+        long len = fs.getFileStatus(p).getLen();
+        ByteBuffer buf = ByteBuffer.allocate((int) len);
+        FSDataInputStream inputStream = fs.open(p);
+        inputStream.readFully(0, buf.array(), 0, buf.array().length);
+        return new String(buf.array());
+    }
+
+    public int dataImport(JavaSparkContext jsc) throws Exception {
+        try {
+            // Get commit time.
+            String commitTime = HoodieActiveTimeline.getNewCommitTime();
+
+            //Get schema.
+            FileSystem fs = FSUtils.getFs();
+            String schemaStr = getSchema(fs);
+
+            // Initialize target hoodie table.
+            Properties properties = new Properties();
+            properties.put(HoodieTableConfig.HOODIE_TABLE_NAME_PROP_NAME, cfg.tableName);
+            properties.put(HoodieTableConfig.HOODIE_TABLE_TYPE_PROP_NAME, cfg.tableType);
+
+            HoodieTableMetaClient.initializePathAsHoodieDataset(fs, cfg.targetPath, properties);
+
+            Path folderPath = new Path(cfg.srcPath);
+            RemoteIterator<LocatedFileStatus> locatedFileStatusRemoteIterator = fs
+                .listFiles(folderPath, true);
+            SQLContext sqlContext = new SQLContext(jsc);
+            JavaRDD<WriteStatus> writeStatuses = jsc.emptyRDD();
+            HoodieWriteClient client = createHoodieClient(jsc, cfg.targetPath, schemaStr, cfg.parallelism);
+
+            Job job = Job.getInstance(jsc.hadoopConfiguration());
+            ParquetInputFormat.setReadSupportClass(job, (AvroReadSupport.class));
+
+            JavaRDD<GenericRecord> genericRecords = jsc
```

Comment:
We actually tried using SQLContext earlier. But there is no tool available for converting Row to GenericRecord. So Instead of having manual conversion we decided to use AvroReader. Do you think we should rather have our own conversion tool for it?